conjur-api/Client.Methods.cs

```diff
@@ -24,6 +28,33 @@ public Variable Variable(string name)
             return new Variable(this, name);
         }
 
+        /// <summary>
+        /// Search for variables
+        /// </summary>
+        /// <param name="query">Query for search.</param>
+        /// <param name="actingAs">Fully-qualified Conjur ID of a role to act as.</param>
+        /// <returns>Returns IEnumerable to VariableInfo.</returns>
+        public IEnumerable<Variable> SearchVariables(string query = null, string actingAs = null)
+        {
+            uint offset = 0;
+            List<SearchVariableResult> searchVarsResult;
+            do
+            {
+                string urlWithParams = $"authz/{GetAccountName()}/resources/variable?offset={offset}"
+                                      + $"&limit={limitSearchVarListsReturned}"
+                                      + ((query != null) ? $"&search={query}" : string.Empty)
+                                      + ((actingAs != null) ? $"&acting_as={actingAs}" : string.Empty);
```

Comment:
I think the `acting_as` part won't work as we need to mention in the value the account & resource type (i.e "MyOrg:user:actingUser"). The account we have internally with GetAccountName() but we need to get the resource type from the calling user (whether it's in the method or in the Client init)

Reply:
`acting_as` is unrelated to this particular function. It should be a method of `Client`, so you can do eg.
```csharp
var aliceClient = client.ActingAs(client.User("alice"));
var rs = aliceClient.ListResources(...);
```

Reply:
Seems fine

Reply:
IMO, creating new instance 'aliceClient' by cloning 'client', seems like over complicating of the code and it is not clear whether a new instance should be created to represent acting_as. 'aliceClient' has some dependency on 'client' login and authentication. Should both share the same credential instance or clone it as well?

Reply:
On second thought, I'm with Moti here. I guess that you meant to have a new Client class member named "kind" which will be edited accordingly in the client.User() (or client.Group() etc.) and then the client will just call the other methods (ListResources() in our case). But what happens in these methods? Do we clone the credentials too? If so, how will the method pass the fact that we're calling this acting as the other user?

In the API, we receive the acting_as as a query param, why should we change the behavior? Are you suggesting not to use the acting_as param in the query at all?